cogs/source/source.py

```diff
@@ -23,37 +25,46 @@ def messageBuilder(titleEnglish: str, anilistID: str, episode: str, similarity:
     return message
 
 
-async def postSourceFunction(ctx, imageURL):
-    """helper method"""
-    try:
-        # use the API to get results
+class Source(commands.Cog):
 
-        tracemoe = tracemoepy.AsyncTrace()
-        result = await tracemoe.search(imageURL.strip("<>"), is_url=True)
-        titleEnglish = result.result[0].anilist.title.english or "No Title Found"
-        anilistID = f"{result.result[0].anilist.id}" or "No anilistID Found"
-        episode = f"{result.result[0].episode}" or "No Episode Found"
-        similarity = float(result.result[0].similarity) or 0
-        await tracemoe.aio_session.close()
+    tracemoe = None
 
-        # send the message using messageBGuilder to build the message
-        await ctx.send(messageBuilder(titleEnglish, anilistID, episode, similarity))
+    def __init__(self, bot: Red):
+        self.tracemoe = tracemoepy.AsyncTrace()
```

Comment:
Can we save `bot here and use it to create task to close session later:
```suggestion
        self.bot = bot
        self.logger = logging.getLogger("red.eilacogs.Source")
        self.tracemoe = tracemoepy.AsyncTrace()
```